### PR TITLE
BUGFIX: Improve error of nearest content collection

### PR DESCRIPTION
**Pull request by mhsdesign**
Currently in 9.0-dev
> Fatal error: Uncaught TypeError: Neos\Neos\Fusion\Helper\NodeHelper_Original::nearestContentCollection(): Argument https://github.com/neos/neos-development-collection/pull/2 ($nodePath) must be of type string, null given

will be thrown, while in 8.3 the error was more readable

**Upgrade instructions**

**Review instructions**

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

File: Neos.Neos/Classes/Fusion/Helper/NodeHelper.php [hunk 1]
```diff
@@ -193,6 +144,59 @@ public function subgraphForNode(Node $node): ContentSubgraphInterface
...
+                    . ' through the "flow structureadjustments:fix --node-type %s" command.',
+                    $contentCollectionType,
+                    $nodePathOfNode->value,
+                    $nodePath->serializeToString(),
+                    $node->nodeTypeName->value
+                ), 1389352984);
+            }
+        }
+    }
+
```
Comment (bwaidelich) [inline Neos.Neos/Classes/Fusion/Helper/NodeHelper.php:199]: just wondering: why did you move this? makes it quite hard to review
I diffed it locally and the change looks fine to me:
image

**Review by bwaidelich — APPROVED** [review]
+1 by reading

File: Neos.Neos/Classes/Fusion/Helper/NodeHelper.php (hunk 1 above)
Comment (mhsdesign) [reply Neos.Neos/Classes/Fusion/Helper/NodeHelper.php:199]: thanks for going the extra route and not trusting any change (id do the same;)) (controlle ist gut und so...)
the idea was that this, being an internal helper, should be lower in visibility and rather be placed at the end.
Especially that with Neos 9 the `Neos.Node` helper will be the central place to access stuff that was previously on the node and will be more noticed (in 8.3 the helper contained mostly just hacky random stuff).

File: Neos.Neos/Classes/Fusion/Helper/NodeHelper.php [hunk 2]
```diff
@@ -193,6 +144,59 @@ public function subgraphForNode(Node $node): ContentSubgraphInterface
...
     }
 
+    /**
+     * Check if the given node is already a collection, find collection by nodePath otherwise, throw exception
+     * if no content collection could be found
+     *
+     * @throws Exception
+     * @internal implementation detail of Neos.Neos:ContentCollection
+     */
+    public function nearestContentCollection(Node $node, ?string $nodePath): Node
```
Comment (kitsunet) [inline Neos.Neos/Classes/Fusion/Helper/NodeHelper.php:154]: Why don't we just forbid `null` nodePaths here. Also Sneakily added internal, that's no good as this exists since ages and ever was internal.

Comment (mhsdesign) [reply Neos.Neos/Classes/Fusion/Helper/NodeHelper.php:154]: the helper is just a bit weird and mostly (see docs) only exists to improve errors if no content collection could be found.

The main change of this pr was to make it nullable again as otherwise a php type error will be raised instead of a human readable 

> No content collection of type %s could be found in the current node and no node path was provided.'
>                     . ' You might want to configure the nodePath property'
>                     . ' with a relative path to the content collection

The helper doesnt do anything that is not doable via native eel and fusion, that is why i added the `@internal` annotation as it very much looks like just to be a means to an end of the `Neos.Neos:ContentCollection`.
It just seems like a hack an with the `Neos.Node` utility now becoming more important to access node things i rather like to hide the ugly parts:)

Comment (kitsunet) [reply Neos.Neos/Classes/Fusion/Helper/NodeHelper.php:154]: Mmm, yes works for me, lets go with it.

**Review by kitsunet — APPROVED** [review]
